{
	"schema_version": "1.6.0",
	"id": "BIT-tensorflow-2022-36016",
	"modified": "2024-11-27T19:40:48.342Z",
	"published": "2024-03-06T11:12:01.872Z",
	"aliases": [
		"CVE-2022-36016",
		"GHSA-g468-qj8g-vcjc"
	],
	"details": "TensorFlow is an open source platform for machine learning. When `tensorflow::full_type::SubstituteFromAttrs` receives a `FullTypeDef& t` that is not exactly three args, it triggers a `CHECK`-fail instead of returning a status. We have patched the issue in GitHub commit 6104f0d4091c260ce9352f9155f7e9b725eab012. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.",
	"affected": [
		{
			"package": {
				"ecosystem": "Bitnami",
				"name": "tensorflow",
				"purl": "pkg:bitnami/tensorflow"
			},
			"severity": [
				{
					"type": "CVSS_V3",
					"score": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H"
				}
			],
			"ranges": [
				{
					"type": "SEMVER",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "2.7.2"
						},
						{
							"introduced": "2.8.0"
						},
						{
							"fixed": "2.8.1"
						},
						{
							"introduced": "2.9.0"
						},
						{
							"fixed": "2.9.1"
						}
					]
				}
			],
			"database_specific": {
				"source": "https://github.com/bitnami/vulndb/tree/main/data/tensorflow/BIT-tensorflow-2022-36016.json"
			}
		}
	],
	"references": [
		{
			"type": "WEB",
			"url": "https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/math_ops.cc"
		},
		{
			"type": "WEB",
			"url": "https://github.com/tensorflow/tensorflow/commit/6104f0d4091c260ce9352f9155f7e9b725eab012"
		},
		{
			"type": "WEB",
			"url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-g468-qj8g-vcjc"
		}
	],
	"database_specific": {
		"cpes": [
			"cpe:2.3:a:google:tensorflow:*:*:*:*:*:*:*:*"
		],
		"severity": "Medium"
	}
}
